e2e/app/page/google-login.ts

```diff
@@ -50,24 +52,24 @@ export default class GoogleLoginPage extends BasePage {
     // Handle Google "Use another account" dialog if it exists
     const useAnotherAccountXpath = '//*[@role="link"]//*[text()="Use another account"]';
     const elemt1 = await Promise.race([
-      this.page.waitForXPath(SELECTOR.emailInput, {visible: true, timeout: 60000}),
-      this.page.waitForXPath(useAnotherAccountXpath, {visible: true, timeout: 60000}),
+      this._page.waitForXPath(SELECTOR.emailInput, {visible: true, timeout: 60000}),
+      this._page.waitForXPath(useAnotherAccountXpath, {visible: true, timeout: 60000}),
     ]);
 
     // compare to the Use another account link
-    const [link] = await this.page.$x(useAnotherAccountXpath);
-    const isLink = await this.page.evaluate((e1, e2) => e1 === e2, elemt1, link);
+    const [link] = await this._page.$x(useAnotherAccountXpath);
+    const isLink = await this._page.evaluate((e1, e2) => e1 === e2, elemt1, link);
     if (isLink) {
       // click " Use another Account " link
-      await this.clickAndWait(link);
+      await BaseElement.asBaseElement(this._page, link).clickAndWait();
     }
 
     const emailInput = await this.email();
     await emailInput.focus();
     await emailInput.type(userEmail);
 
-    const nextButton = await this.page.waitForXPath(SELECTOR.NextButton);
-    await this.clickAndWait(nextButton);
+    const nextButton = await this._page.waitForXPath(SELECTOR.NextButton);
+    await BaseElement.asBaseElement(this._page, nextButton).clickAndWait();
```

Comment:
are all `BaseElement`s clickable? If you expect a button there as part of the contract you're testing, why not create a `Button`?